<launch>
<!-- This launch file is for policies trained with isaac lab and the model is exported to onnx -->
  <arg name="onnx_path" default="rsl_rl/kingfisher_direct/2025-08-20_16-21-26_final-seed44/exported/policy.onnx"/>
  <arg name="cmd_topic" default="/control_agent/cmd_drive_raw"/>
  <arg name="odom_topic" default="/local_odom"/>
  <arg name="goal_topic" default="/goal"/>
  <arg name="status_topic" default="/agent_status"/>
  <arg name="goal_timeout" default="30"/>

  <node pkg="kingfisher_rl" type="capture_agent_onnx.py" name="rl_agent" output="screen">
    <remap from="~goal" to="$(arg goal_topic)"/>
    <remap from="~odom" to="$(arg odom_topic)"/> <!-- /imu/odometry | /pose_gt -->
    <remap from="~rl_status" to="$(arg status_topic)"/>
    <remap from="~cmd_drive" to="$(arg cmd_topic)"/>

    <param name="onnx_path" value="$(arg onnx_path)"/>
    <param name="dist_threshold" value="0.1"/>
    <param name="control_freq" value="10"/>
    <param name="obs_buffer_len" value="1"/>
    <param name="goal_timeout" value="$(arg goal_timeout)"/>
  </node>

</launch>
